### ISSUE TITLE
Dedicated button for "In Order" vs "One Song" option

### ISSUE THREAD

**Issue by aniqah** [quodlibet/quodlibet#2518]
Hi there,

I've been using Quod Libet since I switched to OSX from Linux. (Thanks for making such a great piece of open-source software!) Apologies if this has been discussed before, but I didn't see an open issue after a brief search.

I frequently switch between playing just one song or the whole playlist in order, but the way the UI is set up, there's no way to see which option is currently selected without clicking on the "^" button. Unless there are additional options I'm not aware of, because there are only two options which are mutually exclusive, it seems weird to have the selected option hidden from view and to require two clicks to change the option.

I propose a button similar to the "restart the playlist when finished" or "toggle shuffle mode" buttons that would be called "play only one song" or something similar, with an appropriate icon. What do you folks think? I haven't contributed to this project before, but I'm comfortable with Python and would be interested in trying to do this, maybe with a bit of guidance.

**Comment by aniqah** [comment]
Sorry, I realised I was using version 3.7.1 instead of the latest, version 3.9.1.

I see that there is no longer just a "repeat" button, but the "^" button hides three options: "repeat all", "repeat this track", and "one song". Instead, I propose that there are two buttons, "repeat" and "one track". The functionality would be something like this:

|  | repeat ON | repeat OFF |
| --- | --- | --- |
| **one track ON** | repeat this track | play this track once |
| **one track OFF** | repeat all | play all once |

I think this would be more intuitive, simple, and visible.

**Comment by declension** [comment]
Yes, the idea of a third button has come up a few times actually (#2482, most recently). Less related: #678.

In fact it never quite fitted in, and that particular aspect didn't change much during the rethink of #1411. It awkwardly straddles the worlds of player controls and Play Orders.

Personally I would be quite happy with another button, and think it probably belongs with the other two, though you could also argue it's part of the player.

**Comment by catmatist** [comment]
I just now played with this in 4.0.2 and the fact that you have to turn Repeat ON in order to make selecting One Song have any effect seems completely insane to me. There should be no connection between the two.

**Comment by catmatist** [comment]
I've been on a quest for a Quodlibet mode that works like the default, except that when it gets to the end of the current song, it either plays the next song from the queue, or if the queue is empty, stops playing. It turns out that this setting exists in 4.0.2, but is hard to find, first because it is pretending to be a repeat mode, and second because it isn't made particularly clear that if songs are in the queue, they will all play even if "one song" repeat mode is on. In fact it is so unclear that I worry that someone will decide it's a bug and "fix" it. Oh, and there is this other mode setting called "Stop when queue is empty" that sounds like what I was looking for, but isn't (see #2801).
